### PR TITLE
[Feature] implement return indexer topk of sparse attention

### PR DESCRIPTION
**Pull request by Ronald1995**
## Purpose
Add an `--enable-return-indexer-topk` flag to vLLM that surfaces per-token
sparse-attention top-k indices (the KV slots selected by the indexer layer)
back to the client alongside generated tokens. This mirrors the existing
`--enable-return-routed-experts` flag and follows the same capture →
D2H → slot-indexed-persist → API-encode pipeline.

please see https://github.com/vllm-project/vllm/issues/47280， new PR https://github.com/vllm-project/vllm/pull/49555
## Test Plan

**Server flag:**
```bash
vllm serve deepseek-ai/DeepSeek-V3.2 \
    --enable-return-indexer-topk ...
```

**SamplingParams (per-request):**
```python
class SamplingParams:
    indexer_topk_prompt_start: int = 0
    # Skip the first N prompt tokens from the returned topk. In multi-turn
    # agent scenarios, set to the length of the already-returned prefix to
    # avoid duplicating topk for prompt tokens covered by earlier turns.
```

**OpenAI Chat/Completion response:**
```json
{
  "choices": [{
    "index": 0,
    "message": { "role": "assistant", "content": "..." },
    "finish_reason": "stop",
    "indexer_topk": "<base64-encoded .npy bytes>"
  }]
}
```

Decode:
```python
import base64, io, numpy as np
arr = np.load(io.BytesIO(base64.b64decode(choice["indexer_topk"])))
# arr.shape == (num_tokens, num_indexer_layers, index_topk), dtype=int32
```

## Test Result

pytest -sv tests/entrypoints/openai/test_return_indexer_topk.py

(EngineCore pid=1449893) WARNING 07-22 18:30:04 [jit_monitor.py:135] Triton kernel JIT compilation during inference: _compute_slot_mapping_kernel. This causes a latency spike; consider extending warmup to cover this shape/config.
(EngineCore pid=1449893) WARNING 07-22 18:30:04 [jit_monitor.py:135] Triton kernel JIT compilation during inference: _convert_req_index_to_global_index_kernel. This causes a latency spike; consider extending warmup to cover this shape/config.
(APIServer pid=1449657) INFO:     127.0.0.1:34808 - "POST /v1/completions HTTP/1.1" 200 OK
PASSED
tests/entrypoints/openai/test_return_indexer_topk.py::test_indexer_topk_chat_completions (APIServer pid=1449657) INFO:     127.0.0.1:34810 - "POST /v1/chat/completions HTTP/1.1" 200 OK
PASSED
tests/entrypoints/openai/test_return_indexer_topk.py::test_indexer_topk_prompt_start_offset (APIServer pid=1449657) INFO:     127.0.0.1:34812 - "POST /v1/completions HTTP/1.1" 200 OK
(APIServer pid=1449657) INFO:     127.0.0.1:34812 - "POST /v1/completions HTTP/1.1" 200 OK
PASSED[RemoteOpenAIServer] Sent SIGTERM to process 1449657
(APIServer pid=1449657) INFO 07-22 18:30:06 [launcher.py:100] [shutdown] API server: shutdown triggered
(APIServer pid=1449657) INFO 07-22 18:30:06 [launcher.py:116] [shutdown] API server: stopping engine client mode=abort timeout=0s
(APIServer pid=1449657) INFO 07-22 18:30:06 [core_client.py:655] [shutdown] MPClient: start timeout=0s
(APIServer pid=1449657) INFO 07-22 18:30:06 [core_client.py:657] [shutdown] MPClient: stopping engine manager
(APIServer pid=1449657) WARNING 07-22 18:30:06 [utils.py:626] [shutdown] Process manager: force killing remaining processes count=1
(EngineCore pid=1449893) INFO 07-22 18:30:06 [core.py:1313] [shutdown] EngineCore: trigger received signal=SIGTERM
(EngineCore pid=1449893) INFO 07-22 18:30:06 [core.py:1432] [shutdown] EngineCore: start mode=abort timeout=0s
(EngineCore pid=1449893) INFO 07-22 18:30:06 [core.py:1463] [shutdown] EngineCore: request processing complete; starting resource teardown
(EngineCore pid=1449893) INFO 07-22 18:30:06 [core.py:1326] [shutdown] EngineCore: exiting busy loop
(APIServer pid=1449657) INFO 07-22 18:30:06 [core_client.py:659] [shutdown] MPClient: engine manager stopped
(APIServer pid=1449657) INFO 07-22 18:30:06 [core_client.py:660] [shutdown] MPClient: cleaning up background resources
(APIServer pid=1449657) INFO 07-22 18:30:06 [core_client.py:662] [shutdown] MPClient: complete
(APIServer pid=1449657) INFO:     Shutting down
(APIServer pid=1449657) INFO 07-22 18:30:06 [launcher.py:125] [shutdown] API server: engine client stopped
(APIServer pid=1449657) INFO 07-22 18:30:06 [launcher.py:128] [shutdown] API server: signalling HTTP server shutdown
(APIServer pid=1449657) INFO 07-22 18:30:06 [launcher.py:149] [shutdown] API server: shutting down FastAPI HTTP server
(APIServer pid=1449657) INFO:     Shutting down
(APIServer pid=1449657) INFO:     Waiting for application shutdown.
(APIServer pid=1449657) INFO:     Application shutdown complete.
[RemoteOpenAIServer] Server 1449657 terminated gracefully
[RemoteOpenAIServer] GPU memory released to 0.51 GB (target: 2.65 GB) in 0.0s

=============================================================================================== warnings summary ================================================================================================
<frozen importlib._bootstrap>:241
  <frozen importlib._bootstrap>:241: DeprecationWarning: builtin type SwigPyPacked has no __module__ attribute

<frozen importlib._bootstrap>:241
  <frozen importlib._bootstrap>:241: DeprecationWarning: builtin type SwigPyObject has no __module__ attribute

../../../data/miniconda3/envs/liurong_vllm/lib/python3.11/site-packages/torch/jit/_script.py:365: 14 warnings
  /data/miniconda3/envs/liurong_vllm/lib/python3.11/site-packages/torch/jit/_script.py:365: DeprecationWarning: `torch.jit.script_method` is deprecated. Please switch to `torch.compile` or `torch.export`.
    warnings.warn(

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
======================================================================================== 3 passed, 16 warnings in 58.73s ========================================================================================
sys:1: DeprecationWarning: builtin type swigvarlink has no __module__ attribute
---

<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [ ] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan, such as providing test command.
- [ ] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
</details>

### CODE REVIEW

**Comment by aoshen02** [comment]
Hi, as the stuff of "xx replay" becomes bigger and bigger, to make it production-ready, we don't plan to return it via http endpoint, we will put it in mooncake storage.

**Comment by Ronald1995** [comment]
> Hi, as the stuff of "xx replay" becomes bigger and bigger, to make it production-ready, we don't plan to return it via http endpoint, we will put it in mooncake storage.

okay，i will follow your new plan to implement this feature.

**Comment by Ronald1995** [comment]
close it because of github status problem, new pr https://github.com/vllm-project/vllm/pull/49555
